### ISSUE TITLE
Learning: Article for beginners

### ISSUE THREAD

**Issue by ilatif** [mozilla/pdf.js#7915]
Hi,

_I' am really sorry for creating an issue for this. I read the guides and was confused which would be the best place to discuss something like this._

I have written an article on PDF.js for beginners. It covers many topics like downloading the necessary libraries, explaining the role of `WebWorkers`, displaying pdf pages on canvas, displaying pdf pages via SVG, rendering text-layers and how to zoom in/out pdf pages.

You can find the article at https://www.sitepoint.com/custom-pdf-rendering/. It would be awesome if that article could be added in `README`.

If someone from core team approves this article, then I will submit a Pull Request.

Looking forward to hear from you.

Thanks

**Comment by Snuffleupagus** [comment]
> You can find the article at https://www.sitepoint.com/custom-pdf-rendering/. It would be awesome if that article could be added in README.

In my opinion, I don't think that should be including a link to a third-party resource (that we have no control over) in the `README`, since we wouldn't be able to keep that up to date.

Furthermore, having just skimmed through the article, it seems to me that while there may be some good information provided, it also contains a fair amount of information that probably isn't entirely appropriate for beginners (or others). Some very quick observations:

 - It's probably not a great idea to mention to beginners (or anyone really) that workers can be disabled, since doing so generally leads to much worse performance and the risk of locking up the UI.
(The ability to disable workers is mostly intended to support older browsers without proper worker support, and isn't something that should normally be done.)

 - Given that the SVG backend is far from complete, I do *not* think that you should be advertising it too much at this point; especially not to beginners, since there's just to much that's currently unsupported.
(However, note that there's already an example, see [/examples/svgviewer](https://github.com/mozilla/pdf.js/tree/master/examples/svgviewer), and that PR #7738 added an SVG mode to the default viewer.)

 - Some of the links to source files point to old, now outdated, releases which isn't a great idea since code changes over time. It's already quite likely some of the code examples in the article is now broken by recent changes (see also below). A tutorial for beginners should always strive to provide up-to-date information.

 - The section on TextLayers has, in my opinion, quite a few issues.
   It's based on an old version of PDF.js, which means that the example would most likely not work with current PDF.js code.
   It also seems to suggest that people should use (un-built) source files *directly*, instead of building them first (the outdated `text_layer_builder.js` file the article links to still contains old-style preprocessor statements). This is not something that we directly support, and should definitely *not* be recommended for beginners, since it's very easy to encounter issues with missing dependencies that way.
  *Please note:* The easiest way to enable text-selection (and other things like annotations), is actually to make use of the `pageviewer`/`simpleviewer` examples instead. These are based on the, not officially released, viewer APIs; refer to [/examples/components](https://github.com/mozilla/pdf.js/tree/master/examples/components).

To summarize, in my opinion, we should neither link to nor include that article in its entirety.
If you want add additional information to https://github.com/mozilla/pdf.js#learning, e.g. links to new/improved [Wiki articles](https://github.com/mozilla/pdf.js/wiki) or new [examples](https://github.com/mozilla/pdf.js/tree/master/examples/), I'd suggest creating smaller PRs focused on specific areas. That way it's easier to review the proposed changes, to ensure that the information provided is correct (and not already present elsewhere).

**Comment by ilatif** [comment]
Hi @Snuffleupagus,

Thanks for commenting. I wholeheartedly agree with you over most of the points you mentioned.

As far as beginner thing is concerned, a beginner to PDF.js doesn't necessarily mean beginner to JavaScript and its ecosystem. Also, including `SimpleViewer` would have done the job, but it is too difficult to extend the underlying API/behavior and point of the article was to give users the ability to use low-level APIs PDF.js provide to build an extensible solution matching their needs.

I appreciate your feedback. I' am closing the issue.

Thanks
